File: tests/interfile/java/constructor_taint/src/main/java/Store.java
class Store {
    private String data;

    Store(String data) {
        this.data = data;
    }

    void send() {
        // ruleid: test-constructor-taint
        sink(this.data);
    }

    static void sink(String x) {
        System.out.println(x);
    }
}

File: tests/interfile/cpp/constructor_taint/Store.cpp
#include <iostream>
#include <string>

void sink(const std::string &x) {
    std::cout << x;
}

class Store {
    std::string data;
public:
    Store(const std::string &d) {
        this->data = d;
    }
    void send() {
        // ruleid: test-constructor-taint
        sink(this->data);
    }
};
